pkg/build/builder/docker.go

```diff
@@ -416,6 +426,49 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV instruction with the environment variables
+// from env after every FROM instruction in the given Dockerfile instructions
+// and returns a new Dockerfile.
+func insertEnvAfterFrom(instructions []byte, env []kapi.EnvVar) ([]byte, error) {
+	if len(env) == 0 {
+		return instructions, nil
+	}
+	node, err := parser.Parse(bytes.NewReader(instructions))
+	if err != nil {
+		return nil, err
+	}
+
+	// buildEnv is an ENV Dockerfile instruction to define all environment
+	// variables from env.
+	buildEnv := []byte("ENV")
```

Comment:
I know about the case-insensitivity of the Dockerfiles, I also know about the sensitivity of folks for consistency.

Regarding using strings or constants I think we should root for the latest wherever possible and since we already import dockercmd I see no harm to using it or making it less readable. If you cared about readability you wouldn't use bytes but string in the first place, right? ;)